modules/common/service/types.go

```diff
@@ -60,6 +64,30 @@ func (e *Endpoint) String() string {
 	return string(*e)
 }
 
+// Validate - validates if the endpoint is an allowed one.
+func (e *Endpoint) Validate() error {
+	if !slices.Contains([]Endpoint{EndpointInternal, EndpointPublic}, *e) {
+		return fmt.Errorf("invalid endpoint type: %s", e.String())
+	}
+	return nil
+}
+
+// ValidateRoutedOverrides - validates map of RoutedOverrideSpec
+func ValidateRoutedOverrides(basePath *field.Path, overrides map[Endpoint]RoutedOverrideSpec) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// validate the service override key is valid
+	for k := range overrides {
+		path := basePath.Child("override").Child("service").Key(k.String())
```

Comment:
this now assume that all the user of this validation stores the field under .Override.Service . I think is something only the caller knows so we should move this there and do not make assumption about the location of the field in the CR here.

Reply:
right, while we have this in sync across the operators,  we could make it more generic bu passing this in. I'll update it. thanks for the pointer

Reply:
done